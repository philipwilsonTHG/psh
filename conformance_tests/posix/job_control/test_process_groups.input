# Test process group management
# PSH supports process groups for job control (v0.9.0 feature)

# Test basic process creation
echo "Testing basic process creation:"
(echo "Process test") &
basic_pid=$!
echo "Created process"
wait $basic_pid
echo "Basic process completed"

# Test pipeline process group
echo "Testing pipeline process group:"
(echo "pipeline data" | cat | cat) &
pipeline_pid=$!
echo "Pipeline process group created"
wait $pipeline_pid
echo "Pipeline completed"

# Test process substitution
echo "Testing process substitution:"
echo "test data" | cat
echo "Process substitution test completed"

# Test subshell process group
echo "Testing subshell process group:"
(echo "subshell command 1"; echo "subshell command 2") &
subshell_pid=$!
echo "Subshell created"
wait $subshell_pid
echo "Subshell completed"

# Test command substitution process
echo "Testing command substitution process:"
result=$(echo "command substitution test")
echo "Command substitution result: $result"

# Test nested process groups
echo "Testing nested process groups:"
((echo "outer"; (echo "inner"; true) &; wait)) &
nested_pid=$!
echo "Nested process group created"
wait $nested_pid
echo "Nested process groups completed"

# Test process with function
test_process_func() {
    echo "Function in process: $1"
    return 0
}

echo "Testing function in background process:"
(test_process_func "background") &
func_pid=$!
wait $func_pid
echo "Function process completed with status: $?"

# Test process environment
export TEST_ENV_VAR="test_value"
echo "Testing environment in background process:"
(echo "Environment variable: $TEST_ENV_VAR") &
env_pid=$!
wait $env_pid
echo "Environment process completed"

echo "Process group management test completed"