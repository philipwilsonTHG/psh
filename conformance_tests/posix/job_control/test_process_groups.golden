Testing basic process creation:
Created process
Basic process completed
Testing pipeline process group:
Pipeline process group created
Pipeline completed
Testing process substitution:
test data
Process substitution test completed
Testing subshell process group:
Subshell created
Subshell completed
Testing command substitution process:
Command substitution result: command substitution test
Testing nested process groups:
Nested process group created
Nested process groups completed
Testing function in background process:
Function process completed with status: 127
Testing environment in background process:
Environment process completed
Process group management test completed
